packages/batch/src/BatchProcessor.ts

```diff
@@ -121,6 +138,44 @@ class BatchProcessor extends BasePartialBatchProcessor {
       'Not implemented. Use asyncProcess() instead.'
     );
   }
+
+  /**
+   * Parse the record according to the schema passed.
+   *
+   * If the schema is not provided, it returns the record as is.
+   *
+   * @param record The record to be parsed
+   * @param eventType The type of event to process
+   * @param schema The StandardSchema to be used for parsing
+   */
+  public async parseRecord(
```

Comment:
This method strikes me as odd, it's called `parseRecord` but it has a code path where the record is not parsed. I think logically this function should only parse and the conditional on whether to parse or not moves to L115. Something like this:

```ts
const recordToProcess = this.schema == null ? record : (await this.parseRecord(
        record,
        this.eventType,
        this.schema
      ));
```

Reply:
Also, do we expect users to use this function as a standalone (in which case, it should probably be marked as `static` or can it be a private method?